homeassistant/components/hue.py

```diff
@@ -0,0 +1,237 @@
+    ('switch', DISCOVER_SWITCHES, '',),
+    ('sensor', DISCOVER_SENSORS, '',),
+]
+
+HUEBRIDGE = None
+
+# Map ip to request id for configuring
+_CONFIGURING = {}
+
+
+def setup(hass, config):
+    """
+    Setup a Hue bridge
+
+    Set up a 'global' to contain bridge instances to be called.
+    If the bridge is non-local, it cannot be discovered. Multiple bridge can be
+    listed in configuration.
+    Setup discovery
+    """
+
+    # This dict will hold all bridges. The dict key is the bridgeid.
+    # It's much easier to just use 'host', but deals better with multiple IPs
+    # or DHCP swtiching bridge ip?
+    global HUEBRIDGE
+    HUEBRIDGE = {}
+
+    # Fetch config
+    conf = config.get(DOMAIN, {})
+    if not isinstance(conf, list):
+        conf = [conf]
+    for entry in conf:
+        host = entry.get(CONF_HOST, None)
+        if not host:
+            continue
+        # FIXME / This can only happen when discovery beats configuration
+        # loading right?
+        if host in _CONFIGURING:
+            continue
+
+        _LOGGER.info('Attempting bridge setup for %s', host)
+        setup_huebridge(hass, host)
+
+    comp_name = 'light'
+    component = get_component(comp_name)
+    bootstrap.setup_component(hass, component.DOMAIN, config)
+
+    def huebridge_discovered(service, info):
+        """
+        Called when a Hue bridge is discovered.
+        Response from netdisco recovery is an URI, we just need the host/ip
+        ro initialize the phue module
+        """
+        hostname = urlparse(info[1]).hostname
+        setup_huebridge(hass, hostname)
+    discovery.listen(hass, discovery.SERVICE_HUE, huebridge_discovered)
+
+    return True
+
+
+def setup_huebridge(hass, host):
+    """ Setup a phue bridge based on host parameter. """
+    import phue
+
+    # Create bridge object which will try and connect to the referenced host/ip
+    # If we're not allowed (registration exception), start a config request
+    try:
+        bridge = phue.Bridge(
+            host,
+            config_file_path=hass.config.path(PHUE_CONFIG_FILE))
+        bridge_id = bridge.get_api().get('config').get(ATTR_HUE_BRIDGEID)
+    except (
+            ConnectionRefusedError,
+            OSError):
+        _LOGGER.error("Error connecting to the Hue bridge at %s", host)
+        return
+    except phue.PhueRegistrationException:
+        _LOGGER.warning("Connected to Hue at %s but not registered.", host)
+        request_configuration(hass, host)
+        return
+
+    # If we came here and configuring this host, mark as done
+    if host in _CONFIGURING:
+        request_id = _CONFIGURING.pop(host)
+        configurator = get_component('configurator')
+        configurator.request_done(request_id)
+
+    # Add instanciated bridge to our own control global
+    if HUEBRIDGE.get(bridge_id):
+        _LOGGER.info('Bridge %s already initialized, skipping', bridge_id)
+        return
+    HUEBRIDGE[bridge_id] = HueBridge(hass, bridge)
```

Comment:
`HUEBRIDGE` dict key is not the bridge_id. I can also simply use `host`, which doesn't require instantiating a phue instance. Bridge_id is more uniq. At first the idea was to deal with IP changes on bridge end, but that doesn't make sense as phue is initialized with an IP and probably cannot change